app/controllers/users/passwords_controller.rb

```diff
@@ -2,19 +2,29 @@ module Users
   class PasswordsController < Devise::PasswordsController
     include ValidEmailParameter
 
-    before_action :confirm_valid_token, only: [:edit]
-
     def create
-      RequestPasswordReset.new(params[:user][:email]).perform
+      email = params[:user][:email]
+
+      RequestPasswordReset.new(email).perform
+
+      analytics_user = User.find_by_email(email) || NonexistentUser.new
+      analytics.track_event(
+        "Password Reset Requested for #{analytics_user.role} role", analytics_user
+      )
 
       flash[:success] = t('notices.password_reset')
       redirect_to new_user_session_path
     end
 
     def edit
-      resource = User.new
-      resource.reset_password_token = params[:reset_password_token]
-      @password_form = PasswordForm.new(resource)
+      if token_user&.reset_password_period_valid?
```

Comment:
It is Ruby's syntax for the [safe navigation operator](https://en.wikipedia.org/wiki/Safe_navigation_operator). It was introduced in Ruby 2.3.0.